eslintManager.js

```diff
@@ -23,54 +23,61 @@ const fuseOptions = {
 loadAllRules();
 
 function breakOutRuleName(ruleName) {
-    let plugin = null;
-    let key = ruleName;
     let ruleNameParts = ruleName.split('/');
+
+    let scope;
+    if (ruleNameParts[0].startsWith('@')) {
+        [scope, ...ruleNameParts] = ruleNameParts;
+    }
+
+    let plugin;
     if (ruleNameParts.length > 1) {
-        plugin = ruleNameParts[0];
-        key = ruleNameParts[1];
+        [plugin, ...ruleNameParts] = ruleNameParts;
     }
 
-    return {
-        plugin,
-        key
-    };
+    const key = ruleNameParts.join('/');
+
+    return { scope, plugin, key };
 }
 
 function loadAllRules() {
     rules = linter.rules.getAllLoadedRules();
     ruleKeys = Array.from(rules.keys())
         .reduce((ret, item) => {
-            let ruleParts = breakOutRuleName(item);
-            let plugin = ruleParts.plugin || 'base';
+            const { scope, plugin, key } = breakOutRuleName(item);
+            let pluginName = getPluginName(scope, plugin) || 'base';
 
-            ret[plugin] = ret[plugin] || [];
-            ret[plugin].push(ruleParts.key);
+            ret[pluginName] = ret[pluginName] || [];
+            ret[pluginName].push(key);
 
             return ret;
         }, {});
 }
 
-function searchRules({plugin, key}) {
-    const list = ruleKeys[plugin || 'base'] || [];
+function searchRules(pluginName, key) {
+    const list = ruleKeys[pluginName || 'base'] || [];
     const fuse = new Fuse(list, fuseOptions);
     let suggestedRules = fuse.search(key);
 
     return suggestedRules.map(index => {
-        return ((plugin ? plugin + '/' : '') + list[index]);
+        return ((pluginName ? pluginName + '/' : '') + list[index]);
     });
 }
 
-function importPlugin(pluginName) {
-    if (pluginName === null || pluginsImported.indexOf(pluginName) > -1) {
+function importPlugin(scope, plugin) {
+    const pluginName = getPluginName(scope, plugin);
+
+    if (!pluginName || pluginsImported.indexOf(pluginName) > -1) {
         return Promise.resolve(pluginName);
     }
 
     if (pluginsBeingImported.has(pluginName)) {
         return pluginsBeingImported.get(pluginName);
     }
 
-    let pluginLoader = vscode.workspace.findFiles(`**/node_modules/${constants.eslintPluginPrefix}${pluginName}/package.json`, null, 1)
+    const packagePattern = `**/node_modules/${getPluginPackageName(scope, plugin)}/package.json`;
```

Comment:
For increased readability, Could you make this change:

```suggestion
    const pluginPackageName = getPluginPackageName(scope, plugin);
    const packagePattern = `**/node_modules/${pluginPackageName}/package.json`;
```